src/Type/Accessory/NonEmptyArrayType.php

```diff
@@ -76,6 +76,12 @@ public function getConstantStrings(): array
 
 	public function accepts(Type $type, bool $strictTypes): AcceptsResult
 	{
+		if ($type instanceof HasOffsetType
+			|| $type instanceof HasOffsetValueType
+		) {
+			return AcceptsResult::createYes();
+		}
```

Comment:
I am not 100% sure on this fix, as the 2 types do not necessarily mean "offset on array".

did not find a better place to implement it though.

Reply:
I've been sitting on this one because I don't understand what's happening in PHPStan so that this is necessary.

You know how I feel about `instanceof *Type` 😊

It might be nicer to fix this in (HasOffsetType|HasOffsetValueType)::isAcceptedBy, because technically these types are accepted by non-empty-array.

Reply:
ahh great idea. moving the logic into the HasOffset* types resoves my mentioned problem, as I can check for array and non-empty-ness